docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,215 @@
+
+#### Application Services
+
+##### [MessageBus]
+
+Similar to above, the Application Services `MessageBus` configuration will change to allow for secure connection to the Message Bus. The `Username` & `Password` have been replaced with `Authmode` & `SecretPath` for secure connections. See **[Secure Connections](#secure-connections)** section below for details.
+
+```toml
+[MessageBus.Optional]
+    # MQTT Specific options
+    # Client Identifiers
+    ClientId ="<app sevice key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "mqtt"
+```
+
+##### [Binding]
+
+The `Binding` configuration section will require change for the subscribe topic schema describe in the [Message Bus Topics](#message-bus-topics) section above to filter for Events from specific devices. `SubscribeTopic` will change from a string property containing a single topic to the `SubscribeTopics` string property containing a comma separated list of topics. This allows for the flexibility for the property to be a single topic with the `#` wild card so the Application Service receives all Events as it does today.
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/Random-Integer-Device, edgex/events/Random-Boolean-Device"
+```
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/#"
+```
+
+### Secure Connections
+
+As stated earlier,  this ADR is dependent on the  [**Secret Provider for All**](TBD) ADR to provide a common Secret Provider for all Edgex Services to access their secrets. Once this is available, the Message Bus connection can be secured via the following configurable authentications modes which follows the implementation for secure MQTT Export and secure MQTT Trigger used in Application Services.
+
+- **none** - No authentication used
+- **usernamepassword** - Username & password authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **clientcert** - Client certificate and key for authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **cacert** - CA Certificate only, i.e. just a TLS enabled connection
+
+The secrets specified for the above options are pulled from the `Secret Provider` using the configured `SecretPath`. If the `cacert` secret exists along with the secrets for one of the other modes, it will be used in conjunction to enable a TLS connection in addition to the other mode.
```

Comment:
interesting:  thought `clientcert with cacert` should be good enough to avoid `usernamepassword` authentication type of things.

Reply:
Don't conflate the two.  Making a TLS connection and verifying the server-presented certificates are two separate steps.  It is possible (though generally inadvisable) to do the former without the later.

Reply:
yea, originally thought client cert is good for user authentication.

Reply:
Client cert is good enough for user authentication.  But client cert alone doesn't validate the server.

Reply:
Yes, this gives the flexibility to to that.